<a href="https://colab.research.google.com/github/cahecaz/TrabajoFinal_analisisdesentimientos/blob/main/Analisis_Inicial_IA_Tweets.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Trabajo final: analizador de sentimientos

## Etapa 1. Carga y conocimiento del dataset

**Dataset:** IA Tweets Analysis Dataset (Spanish)  
**Fuente:** Zenodo, registro 10821485  
**Objetivo de esta etapa:** cargar una copia original y comprobar su estructura, tipos de datos, valores ausentes, duplicados y distribución de las etiquetas.

> En esta etapa no se modifican los datos originales.

In [ ]:
# 1. Importar la biblioteca necesaria
import pandas as pd

In [ ]:
# 2. Cargar el archivo CSV
# En Google Colab, primero se debe cargar ia_tweets.csv en Archivos.
ruta_archivo = 'https://raw.githubusercontent.com/cahecaz/TrabajoFinal_analisisdesentimientos/main/ia_tweets.csv'

datos = pd.read_csv(ruta_archivo)

In [ ]:
# 3. Visualizar los primeros registros
datos.head()

,ID,text,polarity,favorite_count,retweet_count,user_followers_count,user_friends_count,user_favourites_count,user_statuses_count,user_verified,user_has_extended_profile,user_is_translator,user_protected,user_default_profile
0,0,Comentaba en una charla sobre IA que el proble...,N,21,7,15936,2395,66945,68578,0,1,0,0,0
1,1,"@alvaropons Eso es imposible. Y sí, va a gener...",N,1,0,685,851,169490,128555,0,1,0,0,1
2,2,@alvaropons Lo disruptivo es que tras años de ...,NEU,1,0,685,851,169490,128555,0,1,0,0,1
3,3,"@alvaropons ¿""Prohibición del uso comercial de...",N,0,0,4877,538,6123,3156,0,1,0,0,0
4,4,@santoroydonoso Cómo se está estudiando prohib...,NEU,0,0,15936,2395,66945,68578,0,1,0,0,0


In [ ]:
# 4. Conocer la cantidad de filas y columnas
print('Filas:', datos.shape[0])
print('Columnas:', datos.shape[1])

Filas: 4038
Columnas: 14


In [ ]:
# 5. Conocer las columnas y sus tipos de datos
datos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4038 entries, 0 to 4037
Data columns (total 14 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   ID                         4038 non-null   int64 
 1   text                       4038 non-null   object
 2   polarity                   4038 non-null   object
 3   favorite_count             4038 non-null   int64 
 4   retweet_count              4038 non-null   int64 
 5   user_followers_count       4038 non-null   int64 
 6   user_friends_count         4038 non-null   int64 
 7   user_favourites_count      4038 non-null   int64 
 8   user_statuses_count        4038 non-null   int64 
 9   user_verified              4038 non-null   int64 
 10  user_has_extended_profile  4038 non-null   int64 
 11  user_is_translator         4038 non-null   int64 
 12  user_protected             4038 non-null   int64 
 13  user_default_profile       4038 non-null   int64 
dtypes: int64

In [ ]:
# 6. Contabilizar valores ausentes
datos.isnull().sum()

,0
ID,0
text,0
polarity,0
favorite_count,0
retweet_count,0
user_followers_count,0
user_friends_count,0
user_favourites_count,0
user_statuses_count,0
user_verified,0


In [ ]:
# 7. Contabilizar filas duplicadas y textos repetidos
print('Filas completamente duplicadas:', datos.duplicated().sum())
print('Textos repetidos:', datos['text'].duplicated().sum())

Filas completamente duplicadas: 0
Textos repetidos: 57


In [ ]:
# 8. Examinar la distribución de las etiquetas
distribucion = datos['polarity'].value_counts()
distribucion

,count
polarity,
NEU,2689
P,757
N,592


In [ ]:
# 9. Mostrar la distribución porcentual
porcentaje = datos['polarity'].value_counts(normalize=True).mul(100).round(2)
porcentaje

,proportion
polarity,
NEU,66.59
P,18.75
N,14.66


## Resultados comprobados

- El dataset contiene **4.038 registros y 14 columnas**.
- No presenta valores ausentes ni textos vacíos.
- Los 4.038 identificadores son únicos.
- No existen filas completamente duplicadas, pero hay **57 textos repetidos**.
- La distribución es: **2.689 neutros (66,59 %), 757 positivos (18,75 %) y 592 negativos (14,66 %)**.
- La clase neutra predomina, por lo que el desequilibrio deberá considerarse al dividir los datos y al interpretar precisión, recall y F1-score.
- La decisión sobre el tratamiento de los 57 textos repetidos se adoptará en la etapa de preparación, manteniendo intacto el archivo original.

## Sustento y referencias de esta etapa

La fuente original documenta la procedencia, estructura y uso previsto del dataset (Guerrero-Contreras et al., 2024a). El estudio asociado analiza sentimientos en redes sociales mediante texto y metadatos (Guerrero-Contreras et al., 2024b). Para el posterior análisis del rendimiento por clase se considera que una sola medida puede ocultar diferencias entre categorías, por lo que se conservarán precisión, recall y F1-score por clase (Sokolova & Lapalme, 2009).

### Referencias APA 7

Guerrero-Contreras, G., Balderas-Díaz, S., Serrano-Fernández, A., & Muñoz, A. (2024a). *IA Tweets Analysis Dataset (Spanish)* (Version v1) [Data set]. Zenodo. https://doi.org/10.5281/zenodo.10821485

Guerrero-Contreras, G., Balderas-Díaz, S., Serrano-Fernández, A., & Muñoz, A. (2024b). Enhancing sentiment analysis on social media: Integrating text and metadata for refined insights. En *2024 International Conference on Intelligent Environments (IE)* (pp. 62–69). IEEE. https://doi.org/10.1109/IE61493.2024.10599899

Sokolova, M., & Lapalme, G. (2009). A systematic analysis of performance measures for classification tasks. *Information Processing & Management, 45*(4), 427–437. https://doi.org/10.1016/j.ipm.2009.03.002